pkg/operator/ceph/object/s3-handlers.go

```diff
@@ -71,6 +71,40 @@ func NewS3Agent(accessKey, secretKey, endpoint string, debug bool, tlsCert []byt
 	}, nil
 }
 
+func NewInsecureS3Agent(accessKey, secretKey, endpoint string, debug bool) (*S3Agent, error) {
+	const cephRegion = "us-east-1"
+
+	logLevel := aws.LogOff
+	if debug {
+		logLevel = aws.LogDebug
+	}
+	client := http.Client{
+		Timeout: HttpTimeOut,
+	}
+	client.Transport = &http.Transport{
+		// #nosec G402 api used for testing purpose
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
Is this line the only difference between the `NewS3Agent()` and `NewInsecureS3Agent()` implementation? Could you refactor this so both methods call the same helper to avoid duplication?